R/nested_model_fit.R

```diff
@@ -0,0 +1,134 @@
+#'
+#' @seealso [axe-model_fit]
+#'
+#' @return Axed nested_model_fit object.
+#'
+#' @examplesIf rlang::is_installed(c("parsnip", "nestedmodels"))
+#'
+#' library(nestedmodels)
+#' library(parsnip)
+#'
+#' model <- linear_reg() %>%
+#'   set_engine("lm") %>%
+#'   nested()
+#'
+#' nested_data <- tidyr::nest(example_nested_data, data = -id)
+#'
+#' fit <- fit(model, z ~ x + y + a + b, nested_data)
+#'
+#' # Reduce the model size
+#' butcher(fit)
+#'
+#' @name axe-nested_model_fit
+NULL
+
+#' Remove the call.
+#'
+#' @rdname axe-nested_model_fit
+#' @export
+axe_call.nested_model_fit <- function(x, verbose = FALSE, ...) {
+  old <- x
+  x$fit$.model_fit <- purrr::map(
+    x$fit$.model_fit,
+    axe_call,
+    verbose = FALSE,
+    ...
+  )
+
+  all_disabled <- purrr::map(x$fit$.model_fit, attr, "disabled")
+  disabled <- unique(purrr::list_c(all_disabled, ptype = character()))
+  if (length(disabled) == 0) {
+    disabled <- NULL
+  }
+  add_butcher_attributes(x, old, disabled = disabled, verbose = verbose)
+}
+
+#' Remove controls used for training.
+#'
+#' @rdname axe-nested_model_fit
+#' @export
+axe_ctrl.nested_model_fit <- function(x, verbose = FALSE, ...) {
+  old <- x
+  x$fit$.model_fit <- purrr::map(
+    x$fit$.model_fit,
+    axe_ctrl,
+    verbose = FALSE,
+    ...
+  )
+
+  all_disabled <- purrr::map(x$fit$.model_fit, attr, "disabled")
+  disabled <- unique(purrr::list_c(all_disabled, ptype = character()))
+  if (length(disabled) == 0) {
+    disabled <- NULL
+  }
+  add_butcher_attributes(x, old, disabled = disabled, verbose = verbose)
+}
+
+#' Remove the training data.
+#'
+#' @rdname axe-nested_model_fit
+#' @export
+axe_data.nested_model_fit <- function(x, verbose = FALSE, ...) {
+  old <- x
+  x$fit$.model_fit <- purrr::map(
+    x$fit$.model_fit,
+    axe_data,
+    verbose = FALSE,
+    ...
+  )
+
+  all_disabled <- purrr::map(x$fit$.model_fit, attr, "disabled")
+  disabled <- unique(purrr::list_c(all_disabled, ptype = character()))
+  if (length(disabled) == 0) {
+    disabled <- NULL
+  }
+  add_butcher_attributes(x, old, disabled = disabled, verbose = verbose)
+}
+
+#' Remove environments.
+#'
+#' @rdname axe-nested_model_fit
+#' @export
+axe_env.nested_model_fit <- function(x, verbose = FALSE, ...) {
+  old <- x
+  x$fit$.model_fit <- purrr::map(
+    x$fit$.model_fit,
+    axe_env,
+    verbose = FALSE,
+    ...
+  )
+
+  all_disabled <- purrr::map(x$fit$.model_fit, attr, "disabled")
+  disabled <- unique(purrr::list_c(all_disabled, ptype = character()))
+  if (length(disabled) == 0) {
+    disabled <- NULL
+  }
+  add_butcher_attributes(x, old, disabled = disabled, verbose = verbose)
+}
+
+#' Remove fitted values.
+#'
+#' @rdname axe-nested_model_fit
+#' @export
+axe_fitted.nested_model_fit <- function(x, verbose = FALSE, ...) {
+  old <- x
+  x$fit$.model_fit <- purrr::map(
+    x$fit$.model_fit,
+    axe_fitted,
+    verbose = FALSE,
+    ...
+  )
+  all_disabled <- purrr::map(x$fit$.model_fit, attr, "disabled")
```

Comment:
Can you tell me more about how the disabled methods are intended to be found here? I don't seem to find this working for me:

``` r
library(butcher)

model <- nestedmodels::nested(
  parsnip::set_engine(parsnip::linear_reg(), "lm")
)
nested_data <- tidyr::nest(nestedmodels::example_nested_data, data = -id)
fit <- parsnip::fit(model, z ~ x + y + a + b, nested_data)

fit$fit$.model_fit <- purrr::map(
  fit$fit$.model_fit,
  axe_fitted,
  verbose = TRUE
)
#> ✔ Memory released: "80 B"
#> ✖ Disabled: `fitted()` and `summary()`
#> ✔ Memory released: "0 B"
#> ✔ Memory released: "80 B"
#> ✖ Disabled: `fitted()` and `summary()`
#> ✔ Memory released: "0 B"
#> ✔ Memory released: "80 B"
#> ✖ Disabled: `fitted()` and `summary()`
#> ✔ Memory released: "0 B"
#> ✔ Memory released: "80 B"
#> ✖ Disabled: `fitted()` and `summary()`
#> ✔ Memory released: "0 B"
#> ✔ Memory released: "80 B"
#> ✖ Disabled: `fitted()` and `summary()`
#> ✔ Memory released: "0 B"
#> ✔ Memory released: "80 B"
#> ✖ Disabled: `fitted()` and `summary()`
#> ✔ Memory released: "0 B"
#> ✔ Memory released: "80 B"
#> ✖ Disabled: `fitted()` and `summary()`
#> ✔ Memory released: "0 B"
#> ✔ Memory released: "80 B"
#> ✖ Disabled: `fitted()` and `summary()`
#> ✔ Memory released: "0 B"
#> ✔ Memory released: "80 B"
#> ✖ Disabled: `fitted()` and `summary()`
#> ✔ Memory released: "0 B"
#> ✔ Memory released: "80 B"
#> ✖ Disabled: `fitted()` and `summary()`
#> ✔ Memory released: "0 B"
#> ✔ Memory released: "80 B"
#> ✖ Disabled: `fitted()` and `summary()`
#> ✔ Memory released: "0 B"
#> ✔ Memory released: "80 B"
#> ✖ Disabled: `fitted()` and `summary()`
#> ✔ Memory released: "0 B"
#> ✔ Memory released: "80 B"
#> ✖ Disabled: `fitted()` and `summary()`
#> ✔ Memory released: "0 B"
#> ✔ Memory released: "80 B"
#> ✖ Disabled: `fitted()` and `summary()`
#> ✔ Memory released: "0 B"
#> ✔ Memory released: "80 B"
#> ✖ Disabled: `fitted()` and `summary()`
#> ✔ Memory released: "0 B"
#> ✔ Memory released: "80 B"
#> ✖ Disabled: `fitted()` and `summary()`
#> ✔ Memory released: "0 B"
#> ✔ Memory released: "80 B"
#> ✖ Disabled: `fitted()` and `summary()`
#> ✔ Memory released: "0 B"
#> ✔ Memory released: "80 B"
#> ✖ Disabled: `fitted()` and `summary()`
#> ✔ Memory released: "0 B"
#> ✔ Memory released: "80 B"
#> ✖ Disabled: `fitted()` and `summary()`
#> ✔ Memory released: "0 B"
#> ✔ Memory released: "80 B"
#> ✖ Disabled: `fitted()` and `summary()`
#> ✔ Memory released: "0 B"

purrr::map(fit$fit$.model_fit, attr, "disabled")
#> [[1]]
#> NULL
#> 
#> [[2]]
#> NULL
#> 
#> [[3]]
#> NULL
#> 
#> [[4]]
#> NULL
#> 
#> [[5]]
#> NULL
#> 
#> [[6]]
#> NULL
#> 
#> [[7]]
#> NULL
#> 
#> [[8]]
#> NULL
#> 
#> [[9]]
#> NULL
#> 
#> [[10]]
#> NULL
#> 
#> [[11]]
#> NULL
#> 
#> [[12]]
#> NULL
#> 
#> [[13]]
#> NULL
#> 
#> [[14]]
#> NULL
#> 
#> [[15]]
#> NULL
#> 
#> [[16]]
#> NULL
#> 
#> [[17]]
#> NULL
#> 
#> [[18]]
#> NULL
#> 
#> [[19]]
#> NULL
#> 
#> [[20]]
#> NULL
```

<sup>Created on 2023-03-17 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>